ast/check.go

```diff
@@ -880,15 +878,15 @@ func causedByNilType(err *Error) bool {
 
 // ArgErrDetail represents a generic argument error.
 type ArgErrDetail struct {
-	Have []types.Type `json:"have"`
-	Want []types.Type `json:"want"`
+	Have []types.Type   `json:"have"`
+	Want types.FuncArgs `json:"want"`
 }
 
 // Lines returns the string representation of the detail.
 func (d *ArgErrDetail) Lines() []string {
 	lines := make([]string, 2)
 	lines[0] = fmt.Sprint("have: ", formatArgs(d.Have))
-	lines[1] = fmt.Sprint("want: ", formatArgs(d.Want))
+	lines[1] = fmt.Sprint("want: ", d.Want)
 	return lines
```

Comment:
[nit] Also adapted code, but I wonder why one would use this over
```go
	lines[1] = "want: " + formatArgs(d.Want)
```
or `fmt.Sprintf`.